art/defences/preprocessor/spatial_smoothing.py

```diff
@@ -92,22 +91,33 @@ def __call__(self, x, y=None):
         :return: Smoothed sample
         :rtype: `np.ndarray`
         """
-        if len(x.shape) == 2:
+
+        x_ndim = x.ndim
+        if x_ndim == 2:
+            raise ValueError("Feature vectors detected. Smoothing can only be applied to data with spatial dimensions.")
+
+        if x_ndim > 5:
             raise ValueError(
-                "Feature vectors detected. Smoothing can only be applied to data with spatial " "dimensions."
+                "Unrecognized input dimension. Spatial smoothing can only be applied to image and video data."
             )
-        if self.channel_index >= len(x.shape):
-            raise ValueError("Channel index does not match input shape.")
 
-        size = [1] + [self.window_size] * (len(x.shape) - 1)
-        size[self.channel_index] = 1
-        size = tuple(size)
+        filter_size = [self.window_size] * x_ndim
+        # set filter_size at batch and channels indices to 1
+        filter_size[0] = 1
+        filter_size[self.channel_index] = 1
+        # set filter_size at temporal index to 1
+        if x_ndim == 5:
+            temporal_index = 2 if self.channel_index == 1 else 1
```

Comment:
What do think about adding a similar comment here (e.g. # check if NCFHW or NFHWC) and in the docstring for `param x` as in the JpegCompression defence (#412) to show to which formats the following lines belong?